openmicroscopy/openmicroscopy
# removing FastCGI support

This PR removes FastCGI support (deprecated in https://github.com/openmicroscopy/openmicroscopy/pull/4056) in favor of WSGI as FastCGI support is deprecated since version 1.7 and will be removed in Django 1.9.

To resolve conflict with #4245 this PR replace https://github.com/openmicroscopy/openmicroscopy/pull/4259 

To test dependencies:
- unset ``omero.web.application_server``, uninstall Django ``pip uninstall Django`` and test ``bin/omero web help|start|stop|status|config|syncmedia|clearsessions|iis``. You should get error ``Django not installed!``
- install dependencies using ``pip install -r dist/requirements.txt`` (for python 2.6 use ``requirements_python26.txt``)
- uninstall gunicorn ``pip uninstall gunicorn`` and try export nginx config ``bin/omero web config nginx``, you should get errors ``ERROR: FastCGI support was removed in OMERO 5.2. Install Gunicorn...``
- set ``omero.web.application_server wsgi`` and try export apache config ``bin/omero web config apache``, no errors expected
- start web, statics should be copied and you should get message ``You are deploying OMERO.web using apache and mod_wsgi...``.
- install dependences using ``pip install -r dist/requirements.txt`` (for python 2.6 use ``requirements_python26.txt``)

To test config:
- set ``omero.web.application_server 'fastcgi-tcp'`` and test ``bin/omero web help|start|stop|status|config|syncmedia|clearsessions|iis``. You should get error ``Invalid APPLICATION_SERVER ...```
- run ``bin/omero admin diagnostics`` it should show you the same error in OMERO.web section.
- set ``omero.web.application_server development`` and try export nginx/apache config ``bin/omero web config nginx|apache``, you should get errors ``ERROR: configuration mismatch...``
- set ``omero.web.application_server wsgi`` and try export nginx config ``bin/omero web config nginx``, you should get errors ``ERROR: configuration mismatch...``
- unset ``omero.web.application_server`` and try export nginx config ``bin/omero web config apache``, you should get errors ``ERROR: configuration mismatch...``
- then try to export nginx config using old system flag `bin/omero web config nginx --system`` , you should get errors ``ERROR: --system is no longer supported...``

To test startup:
- unset ``omero.web.application_server`` and revert other settings to the defaults
- export nginx/apache config as in https://www.openmicroscopy.org/site/support/omero5.1/sysadmins/unix/install-web/install-wsgi.html and check if you do not get unexpected warnings errors, etc
- start web as normal ``bin/omero web start``
- check status  ``bin/omero web status``
- stop web  ``bin/omero web stop``
